src/core/lib/transport/handshaker_factory.h

```diff
@@ -35,11 +35,37 @@ namespace grpc_core {
 
 class HandshakeManager;
 
+// Enum representing the priority of the handshakers.
+// The order of the handshakers is decided by the priority.
+// For example kPreTCPConnect handshakers are called before kTCPConnect and so
+// on.
+enum class HandshakerPriority : int {
+  // Handshakers that should be called before a TCP connect. Applicable mainly
+  // for Client handshakers.
+  kPreTCPConnectHandshakers,
+  // Handshakers responsible for the actual TCP connect establishment.
+  // Applicable mainly for Client handshakers.
+  kTCPConnectHandshakers,
+  // Handshakers responsible for the actual HTTP connect established. Applicable
+  // mainly for Client handshakers.
+  kHTTPConnectHandshakers,
+  // Handshakers that should be called before security handshakes but after
+  // connect establishment. Applicable mainly for Server handshakers currently.
+  kReadHeadSecurityHandshakers,
+  // Handshakers that are responsible for post connect security handshakes.
+  // Applicable for both Client and Server handshakers.
+  kSecurityHandshakers,
+};
+
 class HandshakerFactory {
  public:
   virtual void AddHandshakers(const ChannelArgs& args,
                               grpc_pollset_set* interested_parties,
                               HandshakeManager* handshake_mgr) = 0;
+  // Return the priority associated with the handshaker.
+  // The priority can be either HandshakerClientPriority or
+  // HandshakerServerPriority depending on the type of handshaker.
```

Comment:
This comment looks out of date.  There is no `HandshakerServerPriority`.

Reply:
Done, removed.